### ISSUE TITLE
False +ve in inner rule when outer class is inner

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#8928]
## Expected Behavior
```kt
@Test
fun `#8927 - does not inner class report when it's parent is inner class`() {
    val code = """
        class A {
            fun computeExpensive(className: String) = className + "_1"
            // this inner is required as using computeExpensive
            inner class B {
                val bName by lazy { computeExpensive(B::class.java.name) }
                init {
                    print(bName)
                }
                // if parent is inner then below class should be inner
                inner class C {
                    val d = 9
                }
            }
        }
    """.trimIndent()

    assertThat(subject.lintWithContext(env, code)).isEmpty()
}
```

above TC should pass as `inner class C` can't be non inner as it's outer `inner class B` is not inner

## Observed Behavior
Above TC is failing
